cabal-install/tests/UnitTests/Distribution/Client/UserConfig.hs

```diff
@@ -80,6 +83,19 @@ doubleUpdateConfig = bracketTest . const $ do
         listUnique (map show . fromNubList . installSummaryFile $ savedInstallFlags updated)
 
 
+newDefaultConfig :: Assertion
+newDefaultConfig = do
+    sysTmpDir <- getTemporaryDirectory
+    withTempDirectory silent sysTmpDir "cabal-test" $ \tmpDir -> do
+        let configFile  = tmpDir </> "tmp.config"
+        createDirectoryIfMissing True tmpDir
```

Comment:
Either way is fine with me. I will leave it to one of the Cabal maintainers
(e.g. Ryan or Mikhail) to merge.